contracts/carrot-app/src/yield_sources/osmosis_cl_pool.rs

```diff
@@ -35,14 +35,16 @@ pub struct ConcentratedPoolParamsBase<T> {
     // This is not actually a parameter but rather state
     // This can be used as a parameter for existing positions
     pub position_id: Option<u64>,
+    // This is a cache to avoid querying the position information as this query is expensive
+    pub position_cache: Option<FullPositionBreakdown>,
```

Comment:
Again, I'm NEVER saving the structure in state ! This field is just there to store the cache temporarily

Reply:
> Again, I'm NEVER saving the structure in state ! This field is just there to store the cache temporarily

Ah, I see, thanks for clearing that up. Although I would prefer to keep it separately so it doesn't get stored accidentally

Reply:
If I make the storage structure private, it might solve the issue

Reply:
> If I make the storage structure private, it might solve the issue

You can make wrapper around `Strategy` that will have this field(or other useful fields for other protocols) like 
```rust
StrategyFetched{
  strategy: Strategy,
  osmosis_positions: HashSet<position_id, PositionBreakDown>,
  // .. other useful info after fetching strategies
}
```